### ISSUE TITLE
hacking colors and spacing

### ISSUE THREAD

**Issue by dlyr** [nana-4/materia-theme#502]
**Description**

I have two questions (on current master version).
First about sizing, there is  `$medium-size` definition [here](https://github.com/nana-4/materia-theme/blob/b36b379c9256b67e5f63a60c6faf6912f1845b66/src/_variables.scss#L10), but some computation involve constant pixel size (e.g. [here](https://github.com/nana-4/materia-theme/blob/b36b379c9256b67e5f63a60c6faf6912f1845b66/src/gtk/sass/_common-3.20.scss#L294)).
Where does this constants come from ? it raise an issue when I set `$medium-size/6` too low and would like to change with all computed values (e.g. `$medium-size/6`)

Second about colors, if I understand well on-colors and on function, when 
on(dark, primary) will return on-colors['light']['primary']
so my primary color, in dark mode, is init to that value.
Then when I query on(primary), I get on-colors['dark']['primary'] since the tone of primary is evaluated to 'light'.
Is that on purpose ? It seems hard to tweak colors since there are split in two sets that interacts (I resolve the issue by changing the light primary so it give dark tone).

Thanks for your answer.

**Comment by nana-4** [comment]
### About sizing

> Where does this constants come from ?

In that example, `16px` comes from the icon size which is hard-coded, and `8px` comes from our horizontal padding of `entry`. That sizing is to properly place the `image` inside `entry`, and I don't think it's appropriate to use `$medium-size/6` there.

### About colors

> when I query on(primary),

You mean `on($primary)`, not `on(primary)`, right? (When you omit the optional argument `$type` in the `on()` function, you must specify a CSS color instead of a string to the main argument `$color`.)

> I get on-colors['dark']['primary'] since the tone of primary is evaluated to 'light'.

No, you should get on-colors['dark']['**text**'] if the tone of your `$primary` is evaluated to 'light', because the default value of `$type` in the `on()` function is `text`.

> It seems hard to tweak colors since there are split in two sets that interacts (I resolve the issue by changing the light primary so it give dark tone).

I'm not sure what this means. At least our `on()` function should be in line with the Material Design guidelines, I think. If the function does not meet your needs, you also could use your own `$my-color` instead of `on()`.

**Comment by dlyr** [comment]
Thanks a lot, and you're right `on($primary)`. I think I get the point of the `on` function.
The misleading thing for me is that the base `$primary` for light theme is the one defined in on-colors['dark'] and the other way round.
My issue comes from [here](https://github.com/nana-4/materia-theme/blob/b36b379c9256b67e5f63a60c6faf6912f1845b66/src/gtk/sass/apps/_misc.scss#L126) which makes selected text in thunderbird looks odd as shown here :
![Screenshot from 2020-05-13 00-30-12](https://user-images.githubusercontent.com/11328509/81752035-f7c10b00-94b0-11ea-9822-23f7913a3d1a.png)
After changing light primary color to `#288cec` I get
![Screenshot from 2020-05-13 00-33-50](https://user-images.githubusercontent.com/11328509/81752265-74ec8000-94b1-11ea-83d8-d6283a07324e.png)
This behavior do not happens in firefox by the way. I'm not a thunderbird css expert, but it seems that if the colors do not match the light background, it make some kind of swaps between `color` and `background-color`.

And for the sizing, I might have try to reduce size too much ;)

**Comment by nana-4** [comment]
Yeah, in fact the swapping between `color` and `background-color` can be seen in Firefox too. It should be a Thunderbird/Firefox/Gecko specific issue, not a Materia theme issue.

> The misleading thing for me is that the base $primary for light theme is the one defined in on-colors['dark'] and the other way round.

The `dark` colors are still dark, but yeah, I agree the naming is kind of misleading. I'll reconsider about it.

**Comment by dlyr** [comment]
maybe just a comment to state that dark and light colors are global definition, and are both used for dark and light variant.
One last question about geometry: 
what are the minimal values for large, medium and small size for the theme to work great ?
(my current values are 32px, 24px, 14px, and I have tweaked a few padding and margin ...) 

I nearly get what I what, but I find a glitch in nautilus pathbar, when there is only first and last child (without inbetween ...) I have not enough space. This is because I may have reduce some value too much, but the overall headerbar size fits my needs. Here are screenshots, green and red borders are for debugging purpose. After too long time, I do not get which value should I change.
![Screenshot from 2020-05-13 10-41-57](https://user-images.githubusercontent.com/11328509/81791020-785e2680-9506-11ea-974a-ecb7ef45f6b3.png)
![Screenshot from 2020-05-13 10-41-42](https://user-images.githubusercontent.com/11328509/81791025-78f6bd00-9506-11ea-867c-614e176f299d.png)
